R/slide.R

```diff
@@ -559,13 +589,41 @@ get_before_after_from_window <- function(window_size, align, time_type) {
 #'  `epi_slide_mean` and `epi_slide_sum`) take care of window completion
 #'  automatically to prevent associated errors.
 #' @param ... Additional arguments to pass to the slide computation `.f`, for
-#'  example, `algo` or `na.rm` in data.table functions. You don't need to
-#'  specify `.x`, `.window_size`, or `.align` (or `before`/`after` for slider
-#'  functions).
+#'   example, `algo` or `na.rm` in data.table functions. You don't need to
+#'   specify `.x`, `.window_size`, or `.align` (or `before`/`after` for slider
+#'   functions).
+#' @param .prefix Optional [`glue::glue`] format string; name the slide result
+#'   column(s) by attaching this prefix to the corresponding input column(s).
+#'   Some shorthand is supported for basing the output names on `.window_size`
+#'   or other arguments; see "Prefix and suffix shorthand" below.
+#' @param .suffix Optional [`glue::glue`] format string; like `.prefix`. The
+#'   default naming behavior is equivalent to `.suffix =
+#'   "_{.n}{.time_unit_abbr}{.align_abbr}{.f_abbr}"`. Can be used in combination
+#'   with `.prefix`.
+#' @param .new_col_names Optional character vector with length matching the
+#'   number of input columns from `.col_names`; name the slide result column(s)
+#'   with these names. Cannot be used in combination with `.prefix` and/or
+#'   `.suffix`.
+#'
+#' @section Prefix and suffix shorthand:
+#'
+#' [`glue::glue`] format strings specially interpret content within curly
+#' braces. E.g., `glue::glue("ABC{2 + 2}")` evaluates to `"ABC4"`. For `.prefix`
+#' and `.suffix`, we provide `glue` with some additional variable bindings:
+#'
+#' - `{.n}` will be the number of time steps in the computation
+#'    corresponding to the `.window_size`.
+#' - `{.time_unit_abbr}` will be a lower-case letter corresponding to the
+#'     `time_type` of `.x`
+#' - `{.align_abbr}` will be `""` if `.align` is the default of `"right"`;
+#'    otherwise, it will be the first letter of `.align`
+#' - `{.f_abbr}` will be a character vector containing a short abbreviation
+#'    for `.f` factoring in the input column type(s) for `.col_names`
```

Comment:
This can't be in basic-slide-params.R, because it's not shared by `epi_slide`.  It's already shared by {opt,mean,sum} because they are all in one topic.  I'm not adding .prefix and .suffix to `epi[x]_slide` (yet at least), since it's actually a bit distinct: `epi_slide_{opt,mean,sum}` are applying these to input columns to name associated output columns, while `epi[x]_slide` would be applying these to names of columns generated by the user computation.  #534 suggests this feature might be appreciated though.

Because {opt,mean,sum} are all in one topic, the examples are a bit redundant + the mean & sum examples are smashed next to each other without spacing/separation.

todo: fix smashing or just put all the examples in epi_slide_opt and remove redundant ones.

todo: update basic-slide-params.R grouping language

Reply:
basic-slide-params.R already had the appropriate language regarding groupedness of `.x` (that was probably a doc bug before this PR, but this PR makes it true).  I did update it to mention groupedness/ungroupedness of the output.

I reworked the examples, deleting some, reordering others, tweaking discussion, maybe added another.  Could benefit from a sanity check.